lnrpc/routerrpc/router_backend.go

```diff
@@ -800,6 +800,30 @@ func (r *RouterBackend) UnmarshallRoute(rpcroute *lnrpc.Route) (
 	return route, nil
 }
 
+// maybeValidateMPPParams validates that the MPP parameters are compatible with the
+// payment amount. It returns an error if the parameters don't allow the full
+// payment amount to be sent.
+// maybeValidateMPPParams could be enhanced with additional checks
+func maybeValidateMPPParams(amt lnwire.MilliSatoshi, maxParts uint32,
```

Comment:
i think you can just call it `validateMPPParams`

Reply:
also, i think it may be worth adding a test for this. Perhaps there is an itest we can expand on just to make sure the call to SendPaymentV2 fails as expected